### PR TITLE
Update in-memory cache expiry to take into account user-provided value

### LINKED ISSUE #2466: [Bug] AcquireTokenFromClient acquires token from cache after AbsoluteExpirationRelativeToNow
### Microsoft.Identity.Web Library

Microsoft.Identity.Web.TokenCache

### Microsoft.Identity.Web version

2.13.3

### Web app

Sign-in users

### Web API

Protected web APIs (validating tokens)

### Token cache serialization

In-memory caches

### Description

I was testing to find a way to expire a token at a specific timeframe after its generation using AbsoluteExpirationRelativeToNow.
I tested on a 5 minute expiration for the sake of fast results but intend to use a longer timeframe (20 or 22 hours) when actually implementing it

### Reproduction steps

https://github.com/calvinkwtang/CacheTest

### Error message

_No response_

### Id Web logs

_No response_

### Relevant code snippets

```csharp
app.AddInMemoryTokenCache(services =>
{
    services.Configure<MemoryCacheOptions>(options =>
    {
        options.SizeLimit = 500 * 1024;
    });
    services.Configure<MsalMemoryTokenCacheOptions>(options =>
    {
        options.AbsoluteExpirationRelativeToNow = TimeSpan.FromMinutes(5);
    });
});
```

### Regression

_No response_

### Expected behavior

Calls of AcquireTokenFromClient before 5 minutes (based on the code snippet and sample) should return the same token and after 5 minutes should return a new token

### PR DESCRIPTION
**Pull request by pmaytak**
- For in-memory cache, updated the calculation for the cache entry expiry to take into account the user-provided value. Now between the MSAL-suggested expiry and expiry from options (user-provided or default), the shorter time takes precedence (similar to the distributed cache expiry).
- Added unit tests (+ tested manually).
- Update comments for clarity.
- Rename a variable for clarity.

Fixes #2466

### CODE REVIEW

File: src/Microsoft.Identity.Web.TokenCache/Distributed/MsalDistributedTokenCacheAdapterOptions.cs [hunk 1]
```diff
@@ -9,15 +9,15 @@ namespace Microsoft.Identity.Web.TokenCacheProviders.Distributed
 {
     /// <summary>
     /// Options for the MSAL token cache serialization adapter,
-    /// which delegates the serialization to the IDistributedCache implementations
+    /// which delegates the serialization to the <c>IDistributedCache</c> implementations
```
Comment (keegan-caruso) [inline src/Microsoft.Identity.Web.TokenCache/Distributed/MsalDistributedTokenCacheAdapterOptions.cs:12]: nit: generally, I would see this in <see cref=""/>

File: tests/Microsoft.Identity.Web.Test.Common/AssertExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        /// <summary>
+        /// Verifies that a value is within a given range.
+        /// </summary>
+        /// <param name="actual">The actual value to be evaluated</param>
+        /// <param name="expected">The expected middle of the range.</param>
+        /// <param name="variance">The variance below and above the expected value.</param>
+        public static void WithinVariance(TimeSpan actual, TimeSpan expected, TimeSpan variance)
+        {
+            Assert.InRange(actual, expected.Add(-variance), expected.Add(variance));
```
Comment (keegan-caruso) [inline tests/Microsoft.Identity.Web.Test.Common/AssertExtensions.cs:19]: nice!

**Review by keegan-caruso — APPROVED** [review]
LGTM

**Review by JoshLozensky — APPROVED** [review]

**Review by westin-m — APPROVED** [review]

**Review by jennyf19 — APPROVED** [review]
Thanks @pmaytak

**Pull request merged**